### PR TITLE
Add Farmland Trample Event 

### PR DESCRIPTION
**Pull request by UnRealDinnerbone**
This is a more updated version of #3292 for a FarmlandTrampleEvent

### CODE REVIEW

**Comment by mezz** [comment]
Looks good but I don't really see the purpose of splitting into pre/post, I think it should just fire a single event. Even the test mod does not handle the pre/post versions properly.

**Comment by UnRealDinnerbone** [comment]
So should i just make it one event (The Pre event)?

**Comment by mezz** [comment]
Yes, I think that would be best.

**Comment by mezz** [comment]
It may be a good idea to call `entity.canTrample` and pass the result to the event, so subscribers can see what the vanilla result would be.

File: src/main/java/net/minecraftforge/common/ForgeHooks.java [hunk 1]
```diff
@@ -1450,4 +1451,13 @@ else if (item instanceof ItemMonsterPlacer)
         }
         return modId;
     }
+
+    public static boolean onFarmlandTrample(World world, BlockPos pos, IBlockState state, float fallDistance, Entity entity)
+    {
+        boolean trampleResult = entity.canTrample(world, state.getBlock(), pos, fallDistance);
+        BlockEvent.FarmlandTrampleEvent event = new BlockEvent.FarmlandTrampleEvent(world, pos, state, fallDistance, trampleResult, entity);
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/common/ForgeHooks.java]: Why even fire the event if the entity wouldnt normally trample the ground? Is there real use case for making entities more destructive then they already are?

**Comment by LatvianModder** [comment]
This would be useful for FTBUtilities protection

**Comment by UnRealDinnerbone** [comment]
There i updated event to be called only when the farmland is actually trampled

**Comment by mezz** [comment]
Please update this to resolve merge conflicts

**Comment by UnRealDinnerbone** [comment]
Did i just do that wrong?

**Comment by UnRealDinnerbone** [comment]
Update Merge conflicts

**Pull request merged**